urllib3/connection.py

```diff
@@ -299,7 +302,15 @@ def connect(self):
                     'for details.)'.format(hostname)),
                     SubjectAltNameWarning
                 )
-            match_hostname(cert, self.assert_hostname or hostname)
+            try:
+                match_hostname(cert, self.assert_hostname or hostname)
+            except CertificateError as e:
+                log.error(
+                    'Certificate did not match expected hostname: %s. '
+                    'Certificate: %s', asserted_hostname, cert
+                )
+                e.cert = cert
+                raise e
```

Comment:
Other alternative is to just not try to put the `cert` in there and just do `raise` instead of `raise e`. Maybe we don't need the cert in the exception?